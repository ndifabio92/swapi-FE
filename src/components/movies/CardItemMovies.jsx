import React from 'react';
import {Link} from "react-router-dom";
import {Button, Card, CardContent, Typography} from "@mui/material";
import {getImage} from '../../utils/getImage';
import '../styles/cards.css';
import '../styles/img.css';

export const CardItemMovies = ({item}) => {
    const id = item.url.split("/")[5];
    return (
        <Card className="root card">
            <div className="container-img">
                <img className="img-card" src={getImage("films", item.title)} alt={item.name}/>
            </div>
            <CardContent>
                <Typography className="title">{item.title}</Typography>
                <Typography className="title">
                    {item.release_date}
                </Typography>
                <br/>
                <Typography className="description">
                    <span>Director</span> {item.director}
                </Typography>
                <Typography className="description">
                    <span>Producers</span> {item.producer}
                </Typography>
                <br/>
                <Typography className="description">
                    {item.opening_crawl}
                </Typography>
            </CardContent>
            <Button className="btn-character" variant="outlined">
                <Link className="link-character" to={`/films/${id}`}>
                    Read More...
                </Link>
            </Button>
        </Card>

    );
};
