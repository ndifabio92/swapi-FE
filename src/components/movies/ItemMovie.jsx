import React from 'react'
import useApiGetByUrl from '../../hooks/useApiGetByUrl';
import { Accordion, AccordionDetails, AccordionSummary, Typography } from '@mui/material';
import ExpandMoreIcon from '@mui/icons-material/ExpandMore';
import { getImage } from '../../utils/getImage';

import './styles/itemMovie.css';

const ItemMovie = ({ url }) => {
    const { data, loading, error } = useApiGetByUrl(url);
    return (
        <>
            {
                loading ||
                <Accordion className="acordion">
                    <AccordionSummary
                        expandIcon={<ExpandMoreIcon />}
                        aria-controls="panel1a-content"
                        id="panel1a-header"
                    >
                        <Typography className="title">{data.title}</Typography>
                    </AccordionSummary>
                    <AccordionDetails className='acordion-details'>
                        <div className='container-img-description'>
                            <img className="img-card" src={getImage("films", data.title)} alt={data.name} />
                            <>
                                <Typography className="description">
                                    <span>Release Date</span> {data.release_date}
                                </Typography>
                                <Typography className="description">
                                    <span>Director</span> {data.director}
                                </Typography>
                                <Typography className="description">
                                    <span>Producers</span> {data.producer}
                                </Typography>
                            </>
                        </div>
                        <div className='container-opening'>
                            <Typography className="description">
                                {data.opening_crawl}
                            </Typography>
                        </div>
                    </AccordionDetails>
                </Accordion>

            }
        </>
    )
}

export default ItemMovie
