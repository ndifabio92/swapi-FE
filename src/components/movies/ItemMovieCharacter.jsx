import React from 'react'
import useApiGetByUrl from '../../hooks/useApiGetByUrl';
import {Accordion, AccordionDetails, AccordionSummary, Typography} from '@mui/material';
import {getImage} from '../../utils/getImage';
import Loader from "../ui/Loader/Loader";
import {swalAlert} from "../../utils/alert";
import ExpandMoreIcon from "@mui/icons-material/ExpandMore";

import './styles/itemMovie.css';

const ItemMovieCharacter = ({url}) => {
    const {data, loading, error} = useApiGetByUrl(url);
    if (error) swalAlert('error', 'Error', 'The API call attempt failed.');

    return (
        <>
            {
                loading ? <Loader isLoading={loading}/> :
                    <Accordion className="acordion">
                        <AccordionSummary
                            expandIcon={<ExpandMoreIcon/>}
                            aria-controls="panel1a-content"
                            id="panel1a-header"
                        >
                            <Typography className="title">{data.title}</Typography>
                        </AccordionSummary>
                        <AccordionDetails className='acordion-details'>
                            <div className='container-img-description'>
                                <img className="img-card" src={getImage("films", data.title)} alt={data.name}/>
                                <>
                                    <Typography className="description">
                                        <span>Release Date</span> {data.release_date}
                                    </Typography>
                                    <Typography className="description">
                                        <span>Director</span> {data.director}
                                    </Typography>
                                    <Typography className="description">
                                        <span>Producers</span> {data.producer}
                                    </Typography>
                                </>
                            </div>
                            <div className='container-opening'>
                                <Typography className="description">
                                    {data.opening_crawl}
                                </Typography>
                            </div>
                        </AccordionDetails>
                    </Accordion>
            }
        </>
    )
}

export default ItemMovieCharacter
