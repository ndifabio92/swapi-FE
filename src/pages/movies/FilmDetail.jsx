import { lazy, Suspense } from 'react';
import { useParams } from "react-router-dom";
import useApiGetResourceById from "../../hooks/useApiGetResourceById";
import Loader from "../../components/ui/Loader/Loader";
import {Accordion, AccordionDetails, AccordionSummary, Breadcrumbs, Link, Typography} from "@mui/material";
import ExpandMoreIcon from "@mui/icons-material/ExpandMore";
import ListCharacters from "../../components/characters/ListCharacters";
// import BreadcrumbsComponent from "../../components/ui/Breadcrumbs/BreadcrumbsComponent";

const ItemMovie = lazy(() => import('../../components/movies/ItemMovie'));
const TableStarships = lazy(() => import('../../components/starships/TableStarships'));
const TableVehicles = lazy(() => import('../../components/vehicles/TableVehicles'));

const FilmDetail = () => {
    const { id } = useParams();
    const { data, loading, error } = useApiGetResourceById(id, "films");
    return (
        <>
            {
                loading ? <Loader isLoading={loading} />
                    :
                    <div className="container-characters-detail">
                        <div className="container-div">
                            <Breadcrumbs separator=">" aria-label="breadcrumb" style={{color: "white"}}>
                                <Link href="/films" style={{color: "white"}} underline="none">
                                    FILMS
                                </Link>
                                <Typography color="text.primary">
                                    <span>{data?.title.toUpperCase()}</span>
                                </Typography>
                            </Breadcrumbs>
                        </div>
                        <div className="container-div">
                            <Suspense>
                                <ItemMovie url={data.url} />
                            </Suspense>
                        </div>
                        <div className="container-div">
                            {
                                data.characters !== 0 &&
                                <Accordion className="container-character-accordion">
                                    <AccordionSummary
                                        expandIcon={<ExpandMoreIcon />}
                                        aria-controls="panel1a-content"
                                        id="panel1a-header"
                                        className="container-character-accordionSumary"
                                    >
                                        <Typography className="title">Characters</Typography>
                                    </AccordionSummary>
                                    <AccordionDetails>
                                        <Suspense>
                                            <ListCharacters urls={data.characters} />
                                        </Suspense>
                                    </AccordionDetails>
                                </Accordion>
                            }
                        </div>
                        <div className="container-div">
                            {
                                data.starships.length !== 0 &&
                                <Accordion className="container-character-accordion">
                                    <AccordionSummary
                                        expandIcon={<ExpandMoreIcon />}
                                        aria-controls="panel1a-content"
                                        id="panel1a-header"
                                        className="container-character-accordionSumary"
                                    >
                                        <Typography className="title">Starships</Typography>
                                    </AccordionSummary>
                                    <AccordionDetails>
                                        <Suspense>
                                            <TableStarships urls={data.starships} />
                                        </Suspense>
                                    </AccordionDetails>
                                </Accordion>
                            }
                        </div>
                        <div className="container-div">
                            {
                                data.vehicles.length !== 0 &&
                                <Accordion className="container-character-accordion">
                                    <AccordionSummary
                                        expandIcon={<ExpandMoreIcon />}
                                        aria-controls="panel1a-content"
                                        id="panel1a-header"
                                        className="container-character-accordionSumary"
                                    >
                                        <Typography className="title">Vehicles</Typography>
                                    </AccordionSummary>
                                    <AccordionDetails>
                                        <Suspense>
                                            <TableVehicles urls={data.vehicles} />
                                        </Suspense>
                                    </AccordionDetails>
                                </Accordion>
                            }
                        </div>
                    </div>
            }
        </>
    )
}

export default FilmDetail;