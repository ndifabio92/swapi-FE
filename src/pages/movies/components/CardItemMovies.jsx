import React from 'react';
import {Card, CardContent, Typography} from "@mui/material";
import {getImage} from '../../../utils/getImage';
import '../styles/index.css';


export const CardItemMovies = ({item}) => {
    return (
        <Card className="root card">
            <img className="img-films" src={getImage("films", item.title)} alt={item.title}/>
            <CardContent>
                <Typography className="title">{item.title}</Typography>
                <Typography className="title">
                    {item.release_date}
                </Typography>
                <br/>
                <Typography className="description">
                    <span style={{color: "yellow"}}>Director</span> {item.director}
                </Typography>
                <Typography className="description">
                    <span style={{color: "yellow"}}>Producers</span> {item.producer}
                </Typography>
                <br/>
                <Typography className="description">
                    {item.opening_crawl}
                </Typography>
            </CardContent>
        </Card>
    );
};
