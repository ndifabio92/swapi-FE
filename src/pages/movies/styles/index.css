.root {
  max-width: 400px;
  margin: 20px;
  display: flex;
  flex-direction: column;
  align-items: center;
  background: black !important;
  padding: 35px 19px 19px 19px;
}

.title {
  font-size: 25px !important;
  font-weight: bold;
  color: yellow;
  text-align: center;
}

.description {
  font-size: 12px;
  margin-top: 10px;
  color: white;
  text-align: center;
}

.container-list {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
}

section {
  display: flex;
  width: auto;
  justify-content: center;
}

/* section p:first-of-type {
  text-align: center;
  color: yellow;
} */

/* section div {
  width: 0px;
  flex-grow: 1;
  object-fit: cover;
  opacity: 0.8;
  transition: 0.5s ease;
} */
/* 
section div:hover {
  width: 300px;
  opacity: 1;
  filter: contrast(120%);
} */
