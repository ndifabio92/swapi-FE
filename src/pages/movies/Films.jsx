import React from "react";
import { ListComponent } from "../../components/listComponent/ListComponent";
import { CardItemMovies } from "../../components/movies/CardItemMovies";

const Films = () => {
    return (
        <ListComponent
            resource="films"
            CardComponent={CardItemMovies}
        />
    )
}

export default Films;