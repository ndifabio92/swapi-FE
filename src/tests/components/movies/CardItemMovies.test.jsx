import React from 'react';
import {render, screen} from "@testing-library/react";
import {CardItemMovies} from "../../../components/movies/CardItemMovies";
import {MemoryRouter} from "react-router-dom";

describe("se testea la generarion del listado de films", () => {
    const item =
        {
            "title": "The Phantom Menace",
            "episode_id": 1,
            "opening_crawl": "Turmoil has engulfed the\r\nGalactic Republic. The taxation\r\nof trade routes to outlying star\r\nsystems is in dispute.\r\n\r\nHoping to resolve the matter\r\nwith a blockade of deadly\r\nbattleships, the greedy Trade\r\nFederation has stopped all\r\nshipping to the small planet\r\nof Naboo.\r\n\r\nWhile the Congress of the\r\nRepublic endlessly debates\r\nthis alarming chain of events,\r\nthe Supreme Chancellor has\r\nsecretly dispatched two Jedi\r\nKnights, the guardians of\r\npeace and justice in the\r\ngalaxy, to settle the conflict....",
            "director": "George Lucas",
            "producer": "Rick McCallum",
            "release_date": "1999-05-19",
            "characters": [
                "https://swapi.dev/api/people/2/",
            ],
            "planets": [
                "https://swapi.dev/api/planets/1/",
            ],
            "starships": [
                "https://swapi.dev/api/starships/31/",
            ],
            "vehicles": [
                "https://swapi.dev/api/vehicles/33/",
            ],
            "species": [
                "https://swapi.dev/api/species/1/",
            ],
            "created": "2014-12-19T16:52:55.740000Z",
            "edited": "2014-12-20T10:54:07.216000Z",
            "url": "https://swapi.dev/api/films/4/"
        };

    test('debe hacer match con el snapshot', () => {
        const {container} = render(<MemoryRouter><CardItemMovies item={item}/></MemoryRouter>);
        expect(container).toMatchSnapshot();
    });
})