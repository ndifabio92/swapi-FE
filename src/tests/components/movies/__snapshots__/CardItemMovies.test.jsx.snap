// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The generation of the film list is being tested It must match the snapshot 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root root card css-bhp9pd-MuiPaper-root-MuiCard-root"
  >
    <div
      class="container-img"
    >
      <img
        class="img-card"
        src="test-file-stub"
      />
    </div>
    <div
      class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 title css-ahj2mt-MuiTypography-root"
      >
        The Phantom Menace
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 title css-ahj2mt-MuiTypography-root"
      >
        1999-05-19
      </p>
      <br />
      <p
        class="MuiTypography-root MuiTypography-body1 description css-ahj2mt-MuiTypography-root"
      >
        <span>
          Director
        </span>
         
        George Lucas
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 description css-ahj2mt-MuiTypography-root"
      >
        <span>
          Producers
        </span>
         
        Rick McCallum
      </p>
      <br />
      <p
        class="MuiTypography-root MuiTypography-body1 description css-ahj2mt-MuiTypography-root"
      >
        Turmoil has engulfed the
Galactic Republic. The taxation
of trade routes to outlying star
systems is in dispute.

Hoping to resolve the matter
with a blockade of deadly
battleships, the greedy Trade
Federation has stopped all
shipping to the small planet
of Naboo.

While the Congress of the
Republic endlessly debates
this alarming chain of events,
the Supreme Chancellor has
secretly dispatched two Jedi
Knights, the guardians of
peace and justice in the
galaxy, to settle the conflict....
      </p>
    </div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium btn-character css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      <a
        class="link-character"
        href="/films/4"
      >
        Read More...
      </a>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;

exports[`se testea la generarion del listado de films debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root root card css-bhp9pd-MuiPaper-root-MuiCard-root"
  >
    <div
      class="container-img"
    >
      <img
        class="img-card"
        src="test-file-stub"
      />
    </div>
    <div
      class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 title css-ahj2mt-MuiTypography-root"
      >
        The Phantom Menace
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 title css-ahj2mt-MuiTypography-root"
      >
        1999-05-19
      </p>
      <br />
      <p
        class="MuiTypography-root MuiTypography-body1 description css-ahj2mt-MuiTypography-root"
      >
        <span>
          Director
        </span>
         
        George Lucas
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 description css-ahj2mt-MuiTypography-root"
      >
        <span>
          Producers
        </span>
         
        Rick McCallum
      </p>
      <br />
      <p
        class="MuiTypography-root MuiTypography-body1 description css-ahj2mt-MuiTypography-root"
      >
        Turmoil has engulfed the
Galactic Republic. The taxation
of trade routes to outlying star
systems is in dispute.

Hoping to resolve the matter
with a blockade of deadly
battleships, the greedy Trade
Federation has stopped all
shipping to the small planet
of Naboo.

While the Congress of the
Republic endlessly debates
this alarming chain of events,
the Supreme Chancellor has
secretly dispatched two Jedi
Knights, the guardians of
peace and justice in the
galaxy, to settle the conflict....
      </p>
    </div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium btn-character css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      <a
        class="link-character"
        href="/films/4"
      >
        Read More...
      </a>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
